lib/xandra.ex

```diff
@@ -218,11 +218,159 @@ defmodule Xandra do
   @default_address '127.0.0.1'
   @default_port 9042
 
-  @default_start_options [
-    idle_interval: 30_000,
-    default_consistency: :one,
-    protocol_version: :v3
-  ]
+  @options_definition NimbleOptions.new!([
+    address: [
+      type: {:custom, __MODULE__, :__validate_ip_address__, []},
+      doc: """
+        Internal, not to be set by hand.
+      """
```

Comment:
Let's go with this instead:

```suggestion
      doc: false
```